library/src/main/java/org/apache/apex/malhar/lib/window/impl/SpillableSessionWindowedStorage.java

```diff
@@ -52,10 +53,13 @@ public void setup(Context.OperatorContext context)
     if (keyToWindowsMap == null) {
       // NOTE: this will pose difficulties when we try to assign the entries to a time bucket later on.
       // This is logged in APEXMALHAR-2271
-      keyToWindowsMap = scc.newSpillableSetMultimap(bucket, keySerde, (Serde<Window.SessionWindow<K>>)(Serde)windowSerde);
+      // A work around to make session window data never expire and all kept in one time bucket
+      keyToWindowsMap = scc.newSpillableSetMultimap(bucket, keySerde, (Serde<Window.SessionWindow<K>>)(Serde)windowSerde, new TimeExtractor.FixedTimeExtractor(Long.MAX_VALUE));
```

Comment:
The session windows should be kept in one bucket that represents the max time of the given key, so each key will have its session windows in different time buckets. For example, if key A has session windows 3, 5, and 10 (number representing the timestamp), and key B has session windows 1, 2 and 5, then all of key A's session windows belong to the time bucket representing 10, and all of key B's session windows belong to the time bucket 5.